kernels/ZendEngine3/fcall.c

```diff
@@ -420,7 +420,6 @@ int zephir_call_user_function(zval *object_pp, zend_class_entry *obj_ce, zephir_
 	fci.params         = NULL;
 	//fci.params: Passed as separate parameter to prevent the need to convert zval ** to zval *
 	fci.no_separation  = 1;
-	fci.symbol_table   = NULL;
```

Comment:
Just because you saw no impact it doesn't mean there won't be one.
Removing that would be ok if there was no version using it.

Even your own code example shows why it's not a good idea:
`ZEND_ASSERT(fci->symbol_table == NULL);` might fail since `->symbol_table`
might be initialized to a not-null value (due to not-zeroed memory allocation).

So we'll have to introduce ifdef's for 7.1.X here. 
(since that won't land in PHP7.0.X exactly for that backwards-compatibility argument above)